contrib/internal/httptrace/inferred_proxy.go

```diff
@@ -0,0 +1,143 @@
+package httptrace
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal"
+)
+
+const (
+	PROXY_HEADER_SYSTEM = "X-Dd-Proxy"
+	//PROXY_HEADER_START_TIME_MS = "x-dd-proxy-request-time-ms"
+	PROXY_HEADER_START_TIME_MS = "X-Dd-Proxy-Request-Time-Ms"
+	PROXY_HEADER_PATH          = "X-Dd-Proxy-Path"
+	PROXY_HEADER_HTTPMETHOD    = "X-Dd-Proxy-Httpmethod"
+	PROXY_HEADER_DOMAIN        = "X-Dd-Proxy-Domain-Name"
+	PROXY_HEADER_STAGE         = "X-Dd-Proxy-Stage"
+)
+
+type ProxyDetails struct {
+	SpanName  string `json:"spanName"`
+	Component string `json:"component"`
+}
+
+var (
+	supportedProxies = map[string]ProxyDetails{
+		"aws-apigateway": {
+			SpanName:  "aws.apigateway",
+			Component: "aws-apigateway",
+		},
+	}
+)
+
+type ProxyContext struct {
+	RequestTime     string `json:"requestTime"`
+	Method          string `json:"method"`
+	Path            string `json:"path"`
+	Stage           string `json:"stage"`
+	DomainName      string `json:"domainName"`
+	ProxySystemName string `json:"proxySystemName"`
+}
+
+func extractInferredProxyContext(headers http.Header) *ProxyContext {
+	//proxyContent := make(map[string][]string)
+
+	_, exists := headers[PROXY_HEADER_START_TIME_MS]
+	if !exists {
+		println("no proxy header start time")
+		return nil
+	}
+
+	proxyHeaderSystem, exists := headers[PROXY_HEADER_SYSTEM]
+	if !exists {
+		println("no proxy header system")
+		return nil
+	}
+	if _, ok := supportedProxies[proxyHeaderSystem[0]]; !ok {
+		println("unsupported Proxy header system")
+		return nil
+	}
+
+	// Q: is it possible to have multiple values for any of these http headers??
+	return &ProxyContext{
+		RequestTime:     headers[PROXY_HEADER_START_TIME_MS][0],
+		Method:          headers[PROXY_HEADER_HTTPMETHOD][0],
+		Path:            headers[PROXY_HEADER_PATH][0],
+		Stage:           headers[PROXY_HEADER_STAGE][0],
+		DomainName:      headers[PROXY_HEADER_DOMAIN][0],
+		ProxySystemName: headers[PROXY_HEADER_SYSTEM][0],
+	}
+
+}
+
+func tryCreateInferredProxySpan(headers http.Header, parent ddtrace.SpanContext) ddtrace.SpanContext {
+	println("IN TRYCREATE")
+	println("headers are:")
+	for key, values := range headers {
+		fmt.Printf("Key: %s\n", key)
+		println(key)
+		for _, value := range values {
+			fmt.Printf("  Value: %s\n", value)
+			println(value)
+		}
+	}
+	if headers == nil {
+		println("headers nil")
```

Comment:
Do we want to use println here or should we add better logs that can be useful for debugging in the future for other developers? AKA use `"gopkg.in/DataDog/dd-trace-go.v1/internal/log"` and print with a logger